i3pystatus/github.py

```diff
@@ -1,83 +1,544 @@
+
+        self.condition = threading.Condition()
+        self.thread = threading.Thread(target=self.update_loop, daemon=True)
+        self.thread.start()
+
+    def update_loop(self):
+        try:
+            self.perform_update()
+            while True:
+                with self.condition:
+                    self.condition.wait(self.interval)
+                self.perform_update()
+        except Exception:
+            msg = 'Exception in {thread} at {time}, module {name}'.format(
+                thread=threading.current_thread().name,
+                time=time.strftime('%c'),
+                name=self.__class__.__name__,
+            )
+            self.logger.error(msg, exc_info=True)
+
+    @require(internet)
+    def api_request(self, url):
+        self.logger.debug('Making API request to %s', url)
+        try:
+            with urlopen(url) as content:
+                try:
+                    content_type = dict(content.getheaders())['Content-Type']
+                    charset = re.search(r'charset=(.*)', content_type).group(1)
+                except AttributeError:
+                    charset = 'utf-8'
+                response_json = content.read().decode(charset).strip()
+                if not response_json:
+                    self.logger.debug('JSON response from %s was blank', url)
+                    return {}
+                try:
+                    response = json.loads(response_json)
+                except json.decoder.JSONDecodeError as exc:
+                    self.logger.error('Error loading JSON: %s', exc)
+                    self.logger.debug('JSON text that failed to load: %s',
+                                      response_json)
+                    return {}
+                self.logger.log(5, 'API response: %s', response)
+                return response
+        except Exception as exc:
+            self.logger.error(
+                'Failed to make API request to %s. Exception follows:', url,
+                exc_info=True
+            )
+            return {}
+
+    def detect_status_change(self, response=None):
+        if response is not None:
+            # Compare last update to current and exit without displaying a
+            # notification if one is not needed.
+            if self.previous_status is None:
+                # This is the first time status has been updated since
+                # i3pystatus was started. Set self.previous_status and exit.
+                self.previous_status = response
+                return
+            if response == self.previous_status:
+                # No change, so no notification
+                return
+            self.previous_status = response
+
+        if self.previous_status is None:
+            # The only way this would happen is if we invoked the right-click
+            # event before we completed the initial status check.
+            return
+
+        self.notify_status_change()
```

Comment:
Yep, used to be named differently and I changed the function name but not the ref to it. Good catch!